### PR TITLE
Remove support for obsolete Projectfile

### LINKED ISSUE #142: Drop support for empty and legacy dependencies (Projectfile)
We should print a warning in a patch release when we generate a `shard.yml` file on-the-fly when a project doesn't only has a `Projectfile` (legacy) or nothing at all that support for these cases will be removed in the next MINOR release, and one should create / bug the author that they must define a `shard.yml` file, now.

Shards is mature enough!

### PR DESCRIPTION
**Pull request by Sija**
Fixes #142

### CODE REVIEW

**Comment by Sija** [comment]
@ysbaddaden Is it GTG? My knowledge about the internals is limited thus would be good if you'd review it for any possible side-effects.

File: src/resolvers/git.cr [hunk 1]
```diff
@@ -39,13 +39,8 @@ module Shards
       if file_exists?(refs, SPEC_FILENAME)
         capture("git show #{refs}:#{SPEC_FILENAME}")
       else
```
Comment (ysbaddaden) [inline src/resolvers/git.cr]: I suppose we shouldn't generate an empty `shard.yml` when it's missing anymore, but fail with an error instead.

File: src/resolvers/path.cr [hunk 2]
```diff
@@ -12,14 +12,7 @@ module Shards
...
       else
-        projectfile_path = File.join(local_path, "Projectfile")
-
-        if File.exists?(projectfile_path)
-          contents = File.read(projectfile_path)
-          dependencies = parse_legacy_projectfile_to_yaml(contents)
-        end
-
-        "name: #{dependency.name}\nversion: #{DEFAULT_VERSION}\n#{dependencies}"
+        "name: #{dependency.name}\nversion: #{DEFAULT_VERSION}\n"
```
Comment (ysbaddaden) [inline src/resolvers/path.cr]: ditto

**Comment by Sija** [comment]
@ysbaddaden Ready for re-review.

**Pull request merged**